src/engine/source/builder/src/builders/opmap/opBuilderHelperMap.cpp

```diff
@@ -820,6 +820,73 @@ MapOp opBuilderHelperToInt(const std::vector<OpArg>& opArgs, const std::shared_p
     };
 }
 
+// field: +to_bool/$ref
+MapOp opBuilderHelperToBool(const std::vector<OpArg>& opArgs, const std::shared_ptr<const IBuildCtx>& buildCtx)
+{
+    // Assert expected number of parameters
+    builder::builders::utils::assertSize(opArgs, 1);
+
+    // Parameter type check
+    builder::builders::utils::assertRef(opArgs, 0);
+
+    const auto ref = std::static_pointer_cast<Reference>(opArgs[0]);
+
+    if (buildCtx->validator().hasField(ref->dotPath()))
+    {
+        auto sType = buildCtx->validator().getType(ref->dotPath());
+        if (typeToJType(sType) != json::Json::Type::Number)
+        {
+            throw std::runtime_error(fmt::format("Expected number reference but got reference '{}' of type '{}'",
+                                                 ref->dotPath(),
+                                                 schemf::typeToStr(sType)));
+        }
+    }
+
+    // Format name for the tracer
+    const auto name = buildCtx->context().opName;
+
+    // Tracing messages
+    const std::string successTrace {fmt::format(TRACE_SUCCESS, name)};
+    const std::string failureTrace1 {fmt::format(TRACE_REFERENCE_NOT_FOUND, name, ref->dotPath())};
+    const std::string failureTrace2 {fmt::format("{} -> Reference '{}' is not a number", name, ref->dotPath())};
+
+    return [failureTrace1, failureTrace2, successTrace, ref, runState = buildCtx->runState()](
+               base::ConstEvent event) -> MapResult
+    {
+        auto getNumberAsDouble = [&](std::string_view path) -> std::optional<double>
```

Comment:
Check the JSON wrapper available in the base module. There you'll find the getNumberAsDouble method. Please use it.